packages/components/src/List/Toolbar/ColumnChooserButton/ColumnChooser/ColumnChooserRow/RowCheckbox/RowCheckbox.component.js

```diff
@@ -19,24 +14,26 @@ const RowCheckbox = ({
 	intermediate = false,
 }) => {
 	const onChangeCheckbox = event => {
-		onChange(event.target.checked, label);
+		// Force to pass a checked state in case of intermediate checkbox
+		const checkedState = intermediate ? true : event.target.checked;
+		onChange(checkedState, label);
 	};
 	const describedby = `${id}-${description}`;
 	return locked ? (
 		<Fragment>
-			<Icon name="talend-locked" className={theme('tc-column-chooser-row-locked-icon')} />
+			<SizedIcon name="locker-closed" size="M"></SizedIcon>
```

Comment:
Isn't it the same as writting:

```suggestion
			<SizedIcon name="locker-closed" size="M"/>
```